frontend/projects/dipa-frontend/src/app/modules/gantt/gantt.component.ts

```diff
@@ -22,20 +24,52 @@ export class GanttComponent implements OnInit, AfterContentInit {
   periodStartDate = new Date(2020, 0, 1);
   periodEndDate = new Date(2020, 11, 31);
 
+  periodStartDateArray = [];
+  periodEndDateArray = [];
+  
   constructor(public ganttControlsService: GanttControlsService,
               private timelineService: TimelineService) {  }
 
+
   ngOnInit(): void {
-    this.timelineService.getTaskData()
-    .subscribe((data) => {
-      this.taskData = data;
-    });
-
-    this.timelineService.getMilestoneTaskData()
-    .subscribe((data) => {
-      this.milestoneData = data;
-    });
     
+    pipe(
+      switchMap(() => forkJoin([this.timelineService.getTaskData(),this.timelineService.getMilestoneTaskData()])),
+        map(([tasks,milestones]) => {
+          this.periodStartDate  = Math.min.apply(null,[getMinimumStartDate(tasks), getMinimumStartDate(milestones)]);
+          this.periodEndDate    = Math.max.apply(null,[getMaximumEndDate(tasks), getMaximumEndDate(milestones)]);
+        }
+      )
+    )
```

Comment:
@kago1988  Sehr schön! So in etwa hatte ich mir das vorgestellt. Ich frage mich nur gerade ob `this.taskData` und `this.milestoneData` jetzt noch initialisiert werden. Ich sehe zumindest nicht mehr...